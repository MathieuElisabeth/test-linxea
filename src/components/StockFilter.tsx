import { useState, useCallback } from 'react';
import type { FilterState } from '@/types/stock';

interface StockFilterProps {
  tickers: string[];
  onFilterChange: (filter: FilterState) => void;
  selectedTicker: string | null;
}

export const StockFilter = ({ tickers, onFilterChange, selectedTicker }: StockFilterProps) => {
  const [searchTerm, setSearchTerm] = useState('');

  const handleSearchChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
    const newSearchTerm = event.target.value;
    setSearchTerm(newSearchTerm);
    onFilterChange({ searchTerm: newSearchTerm, selectedTicker });
  }, [onFilterChange, selectedTicker]);

  const handleTickerSelect = useCallback((ticker: string) => {
    onFilterChange({ searchTerm, selectedTicker: ticker });
  }, [searchTerm, onFilterChange]);

  const filteredTickers = tickers.filter(ticker => 
    ticker.toLowerCase().includes(searchTerm.toLowerCase())
  );

  return (
    <div className="stock-filter">
      <div className="search-container">
        <input
          type="text"
          placeholder="Rechercher une action..."
          value={searchTerm}
          onChange={handleSearchChange}
          className="search-input"
        />
      </div>
      
      <div className="ticker-list">
        {filteredTickers.length === 0 ? (
          <div className="no-results">Aucune action trouv√©e</div>
        ) : (
          filteredTickers.map(ticker => (
            <button
              key={ticker}
              onClick={() => handleTickerSelect(ticker)}
              className={`ticker-button ${selectedTicker === ticker ? 'selected' : ''}`}
            >
              {ticker}
            </button>
          ))
        )}
      </div>
    </div>
  );
}; 