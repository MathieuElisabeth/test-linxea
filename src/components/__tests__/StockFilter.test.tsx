import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { StockFilter } from '../StockFilter';

describe('StockFilter', () => {
  const mockTickers = ['AAPL', 'MSFT', 'GOOGL', 'TSLA'];
  const mockOnFilterChange = vi.fn();
  const defaultProps = {
    tickers: mockTickers,
    onFilterChange: mockOnFilterChange,
    selectedTicker: null,
  };

  beforeEach(() => {
    mockOnFilterChange.mockClear();
  });

  it('renders correctly with all tickers', () => {
    const { container } = render(<StockFilter {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with selected ticker', () => {
    const { container } = render(<StockFilter {...defaultProps} selectedTicker="AAPL" />);
    expect(container).toMatchSnapshot();
  });

  it('filters tickers based on search input', async () => {
    render(<StockFilter {...defaultProps} />);
    
    const searchInput = screen.getByPlaceholderText('Rechercher une action...');
    fireEvent.change(searchInput, { target: { value: 'AAPL' } });
    
    expect(screen.getByText('AAPL')).toBeInTheDocument();
    expect(screen.queryByText('MSFT')).not.toBeInTheDocument();
    expect(screen.queryByText('GOOGL')).not.toBeInTheDocument();
    expect(screen.queryByText('TSLA')).not.toBeInTheDocument();
  });

  it('displays a message when no tickers match search', () => {
    render(<StockFilter {...defaultProps} />);
    
    const searchInput = screen.getByPlaceholderText('Rechercher une action...');
    fireEvent.change(searchInput, { target: { value: 'XYZ' } });
    
    expect(screen.getByText('Aucune action trouvÃ©e')).toBeInTheDocument();
    expect(screen.queryByText('AAPL')).not.toBeInTheDocument();
  });

  it('renders with empty tickers list', () => {
    const { container } = render(
      <StockFilter 
        tickers={[]} 
        onFilterChange={mockOnFilterChange} 
        selectedTicker={null} 
      />
    );
    expect(container).toMatchSnapshot();
  });
}); 