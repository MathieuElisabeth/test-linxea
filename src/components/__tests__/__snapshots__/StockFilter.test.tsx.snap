// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StockFilter > renders correctly with all tickers 1`] = `
<div>
  <div
    class="stock-filter"
  >
    <div
      class="search-container"
    >
      <input
        class="search-input"
        placeholder="Rechercher une action..."
        type="text"
        value=""
      />
    </div>
    <div
      class="ticker-list"
    >
      <button
        class="ticker-button "
      >
        AAPL
      </button>
      <button
        class="ticker-button "
      >
        MSFT
      </button>
      <button
        class="ticker-button "
      >
        GOOGL
      </button>
      <button
        class="ticker-button "
      >
        TSLA
      </button>
    </div>
  </div>
</div>
`;

exports[`StockFilter > renders correctly with selected ticker 1`] = `
<div>
  <div
    class="stock-filter"
  >
    <div
      class="search-container"
    >
      <input
        class="search-input"
        placeholder="Rechercher une action..."
        type="text"
        value=""
      />
    </div>
    <div
      class="ticker-list"
    >
      <button
        class="ticker-button selected"
      >
        AAPL
      </button>
      <button
        class="ticker-button "
      >
        MSFT
      </button>
      <button
        class="ticker-button "
      >
        GOOGL
      </button>
      <button
        class="ticker-button "
      >
        TSLA
      </button>
    </div>
  </div>
</div>
`;

exports[`StockFilter > renders with empty tickers list 1`] = `
<div>
  <div
    class="stock-filter"
  >
    <div
      class="search-container"
    >
      <input
        class="search-input"
        placeholder="Rechercher une action..."
        type="text"
        value=""
      />
    </div>
    <div
      class="ticker-list"
    >
      <div
        class="no-results"
      >
        Aucune action trouv√©e
      </div>
    </div>
  </div>
</div>
`;
